apps/Launcher/FolderModel.cpp

```diff
@@ -116,6 +138,36 @@ FolderModel::SortOrder FolderModel::getSortOrder() const
     return _sortOrder;
 }
 
+QStringList FolderModel::getYears() const
+{
+    std::set<int> years;
+    const auto parent = getPathIndex(getRootFolder());
+    const auto numRows = rowCount(parent);
+    for (auto i = 0; i < numRows; ++i)
+    {
+        const auto idx = index(i, 0, parent);
+        const auto dateTime = idx.data(FileModified).toDateTime();
+        const auto year = dateTime.date().year();
+        years.insert(year);
+    }
+
+    auto stringList = _toQStringList(years);
+    if (_sortOrder == SortOrder::Descending)
```

Comment:
Oh that's a good idea, I hadn't thought about that! Unfortunately it seems that the comparator is part of the type and thus cannot (easily) be selected at runtime:
https://stackoverflow.com/questions/11185223
What I could do to avoid an additional sorting operation is to conditionally use set.rbegin(), set.rend() in _toQStringList(). Not sure if that has a good impact on code readability, though.

Reply:
the conditional set.rbegin(), set.rend() in _toQStringList() looks quite good in the end, done!